py/train.py

```diff
@@ -33,7 +33,8 @@ def data_converting(raw_data):
             final_data.append((d['text'],temp))
     return final_data, label_set
 
-def main(output_dir, data_path, n_iter, model=None):
+
+def main(data_path, n_iter, model):
```

Comment:
something's not working quite right here.  We're still passing 4 args from the parse. Also these are named wrong still from the web version.  data_path should be raw_data now, since it's just the actual json content.  model should be model_path

Reply:
I'm only seeing 3 arguments being passed. Also output_dir and model would be the same value so I don't think it's necessary to have both

Reply:
in train.py, lines 88-118 is where the arg parsing happens, that's where the issue is, specifically 118-121 i think. It might be easier to keep 4 args: 3 args could work, but train should also create a model if it's not already there, so you would have an output path to save it, but you might not have an actual model there yet.  I think it would error in that case, so you'd have to handle it,  probably changing something line 42.

Reply:
Ahh I see the parser thing now.  I think it would make more sense to keep the main function reserved for training and then separate creating a new model into a function of its own, though. Thoughts?

Reply:
the purist side of me thinks yes separate func, since it seems like a whole separate path, but since the entire model creation is done in 2 lines it would probably be overkill

Reply:
Right, but if we were to click a 'create new model' button we wouldn't necessarily want to train it right away, just create it.  I think that's reason enough to just make a 2 line create_model() function. @tmarkn be our arbiter and settle the dispute i dont care either way just 2 different approaches

Reply:
Also separating logical blocks of code into discrete components is good practice as it makes unit tests a million times easier

Reply:
That's true, but wait, are you talking about 1. a separate function in the python code, 2. a separate script in the python code, or 3. a separate function in the js code

Reply:
Yea probably all 3

Reply:
now you've got me all for it, but i'm trying to do it and it's not as easy as i thought.  the model creation isn't so bad, but the setup for a new model is a little different. i don't know if there's a way to tell if it's been set up or not beforehand (at the moment it checks if you gave it a model to load, but in our case we always give it a model, it just might be blank).  the optimizer we give it in the update method i think attaches it to it, but eff me if i can figure out how to put it in there.  
long story short, so far creating without training is an effing pain

Reply:
tried one in features/create_script, you can see if it works for you